capstone/src/main/webapp/authenticate.js

```diff
@@ -0,0 +1,89 @@
+var CLIENT_ID = config.CLIENT_ID;
+var API_KEY = config.API_KEY;
+var DISCOVERY_DOCS = ["https://script.googleapis.com/$discovery/rest?version=v1"];
+var SCOPES = 'https://www.googleapis.com/auth/forms';
+
+document.getElementById("auth").appendChild(createLoginButton());
+document.getElementById("auth").appendChild(createLogoutButton());
+var authorizeButton = document.getElementById('authorize_button');
+var signoutButton = document.getElementById('signout_button');
+
+/**
+*  On load, called to load the auth2 library and API client library.
+*/
+function handleClientLoad() {
+    gapi.load('client:auth2', initClient);
+}
+
+/**
+*  Initializes the API client library and sets up sign-in state
+*  listeners.
+*/
+function initClient() {
+    gapi.client.init({
+        apiKey: API_KEY,
+        clientId: CLIENT_ID,
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES
+    }).then(function () {
+        // Listen for sign-in state changes.
+        gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+        
+        // Handle the initial sign-in state.
+        updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        authorizeButton.onclick = handleAuthClick;
+        signoutButton.onclick = handleSignoutClick;
+    }, function(error) {
+        appendPre(JSON.stringify(error, null, 2));
+    });
+}
+
+/**
+*  Called when the signed in status changes, to update the UI
+*  appropriately. After a sign-in, the API is called.
+*/
+function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+        authorizeButton.style.display = 'none';
+        signoutButton.style.display = 'block';
+    } else {
+        authorizeButton.style.display = 'block';
+        signoutButton.style.display = 'none';
+    }
+}
+
+function handleAuthClick() {
+    gapi.auth2.getAuthInstance().signIn();
+}
```

Comment:
This can be inlined instead of being separate.

//above
authorizeButton.onclick = gapi.auth2.getAuthInstance().signIn

Notice there are no parenthesis, because you are setting it equal to the function itself, not calling the function

Reply:
Got it. Need to remember that variables can be functions themselves is a plus too.